fhevm/tee_bit.go

```diff
@@ -0,0 +1,145 @@
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) << uint8(b)))
+		case 2:
+			return uint64((uint8(a) << uint8(b)))
+		case 3:
+			return uint64((uint16(a) << uint16(b)))
+		case 4:
+			return uint64((uint32(a) << uint32(b)))
+		case 5:
+			return (a << b)
+		default:
+			return 0
+		}
+	}, "teeShlRun")
+}
+
+func teeShrRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) >> uint8(b)))
+		case 2:
+			return uint64((uint8(a) >> uint8(b)))
+		case 3:
+			return uint64((uint16(a) >> uint16(b)))
+		case 4:
+			return uint64((uint32(a) >> uint32(b)))
+		case 5:
+			return (a >> b)
+		default:
+			return 0
+		}
+	}, "teeShrRun")
+}
+
+func teeRotlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) << uint8(b)) | (uint8(a) >> (uint8(4) - uint8(b)%4)))
+		case 2:
+			return uint64((uint8(a) << uint8(b)) | (uint8(a) >> (uint8(8) - uint8(b)%8)))
+		case 3:
+			return uint64((uint16(a) << uint16(b)) | (uint16(a) >> (uint16(16) - uint16(b)%16)))
+		case 4:
+			return uint64((uint32(a) << uint32(b)) | (uint32(a) >> (uint32(32) - uint32(b)%32)))
+		case 5:
+			return (a << b) | (a >> (64 - b%64))
+		default:
+			return 0
+		}
+	}, "teeRotl")
+}
+
+func teeRotrRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) >> uint8(b)) | (uint8(a) >> (uint8(4) - uint8(b)%4)))
+		case 2:
+			return uint64((uint8(a) >> uint8(b)) | (uint8(a) >> (uint8(8) - uint8(b)%8)))
+		case 3:
+			return uint64((uint16(a) >> uint16(b)) | (uint16(a) >> (uint16(16) - uint16(b)%16)))
+		case 4:
+			return uint64((uint32(a) >> uint32(b)) | (uint32(a) >> (uint32(32) - uint32(b)%32)))
+		case 5:
+			return (a >> b) | (a << (64 - b%64))
+		default:
+			return 0
+		}
+	}, "teeRotr")
+}
+
+func teeBitAndRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doOperationGeneric(environment, caller, input, runSpan, func(a, b any) any {
+		switch a.(type) {
+		case uint64:
+			return a.(uint64) & b.(uint64)
+		case bool:
+			return a.(bool) && b.(bool)
+		default:
+			return nil
```

Comment:
same here, we should error in default (and all other functions). Probably the `operator` argument should take an error